src/new_index/schema.rs

```diff
@@ -424,6 +424,33 @@ impl ChainQuery {
         }
     }
 
+    pub fn get_headers(&self, from: &BlockHash, count: usize) -> Option<Vec<BlockHeader>> {
+        let _timer = self.start_timer("get_headers");
+
+        if let Some(from_header) = self
+            .store
+            .indexed_headers
+            .read()
+            .unwrap()
+            .header_by_blockhash(from)
+        {
+            Some(
+                self.store
+                    .indexed_headers
+                    .read()
+                    .unwrap()
+                    .iter()
+                    .rev()
+                    .skip(self.best_height() - from_header.height())
```

Comment:
The best height might change between calling `best_height()` and starting the iteration itself. `get_headers` should grab one read lock and reuse it to ensure the state is consistent.